### PR TITLE
Murano as service

### PR DESCRIPTION
**Pull request by hmunfru**
#### Reviewers

@flopezag @jicarretero 
#### Description

Install murano from ubuntu packages
#### Testing

Documentation and docker

### CODE REVIEW

File: docker/muranoservice/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM ubuntu:16.04
```
Comment (flopezag) [inline docker/muranoservice/Dockerfile:8]: Missing description how to launch the docker.

File: docker/muranoservice/README.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
+# Installing murano as service
+## Install murano packages
+
+Enable Openstack repositories:
+
+    apt install software-properties-common
```
Comment (flopezag) [inline docker/muranoservice/README.md]: apt-get or apt?

Anyway add $ befote it.

File: docker/muranoservice/README.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+    apt-get install murano-api murano-engine python-muranoclient
+
+## Create database
+
+To create the database, complete these steps:
+
+Use the database access client to connect to the database server as the root user:
+
+    $ mysql -u root -p
```
Comment (flopezag) [inline docker/muranoservice/README.md:19]: Do we need password? If yes, say it, if not, say it.

File: docker/muranoservice/README.md [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+Create the murano database:
+
+    CREATE DATABASE murano;
+
+Grant proper access to the heat database:
```
Comment (flopezag) [inline docker/muranoservice/README.md]: I just prefer:

Grant proper access to the heat database, replacing _MURANO_DBPASS_ with a suitable password:

File: docker/muranoservice/README.md [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+    GRANT ALL PRIVILEGES ON murano.* TO 'murano'@'localhost' \
+    IDENTIFIED BY 'MURANO_DBPASS';
+
+    GRANT ALL PRIVILEGES ON murano.* TO 'murano'@'%' \
+    IDENTIFIED BY 'MURANO_DBPASS';
+
+Replace MURANO_DBPASS with a suitable password.
```
Comment (flopezag) [inline docker/muranoservice/README.md]: Refer to the previous comment

File: docker/muranoservice/README.md [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+## Install murano packages
+
+Enable Openstack repositories:
+
+    apt install software-properties-common
+    add-apt-repository cloud-archive:newton
+
+Then, install murano packages:
+
+    apt-get install murano-api murano-engine python-muranoclient
```
Comment (flopezag) [inline docker/muranoservice/README.md]: $ apt-get install murano-api murano-engine python-muranoclient

File: docker/muranoservice/README.md [hunk 7]
```diff
@@ -0,0 +1,237 @@
+# Installing murano as service
+## Install murano packages
+
+Enable Openstack repositories:
+
+    apt install software-properties-common
+    add-apt-repository cloud-archive:newton
```
Comment (flopezag) [inline docker/muranoservice/README.md]: $ add-apt-repository cloud-archive:newton

File: docker/muranoservice/README.md [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+     --service-id $(keystone service-list | awk '/ application-catalog / {print $2}') \
+     --publicurl http://controller:8082/v1/%\(tenant_id\)s \
+     --internalurl http://controller:8082/v1/%\(tenant_id\)s \
+     --adminurl http://controller:8082/v1/%\(tenant_id\)s \
+     --region regionOne
+
+## Configure murano
+Edit the /etc/murano/murano.conf file and complete the following actions:
+
+In the [database] section, configure database access:
```
Comment (flopezag) [inline docker/muranoservice/README.md]: In the [database] section, configure database access, replacing MURANO_DBPASS with the password you chose for the Murano database:

File: docker/muranoservice/README.md [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+
+    [database]
+    ...
+    connection = mysql://murano:MURANO_DBPASS@controller/murano
+
+Replace MURANO_DBPASS with the password you chose for the Murano database.
```
Comment (flopezag) [inline docker/muranoservice/README.md]: Take a look to the previous comment.

File: docker/muranoservice/README.md [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+
+In the [DEFAULT] section, configure RabbitMQ message broker access:
```
Comment (flopezag) [inline docker/muranoservice/README.md]: In the [DEFAULT] section, configure RabbitMQ message broker access, replacing RABBIT_PASS with the password you chose for the guest account in RabbitMQ:

File: docker/muranoservice/README.md [hunk 11]
```diff
@@ -0,0 +1,237 @@
...
+
+    [DEFAULT]
+    ...
+    rpc_backend = rabbit
+    rabbit_host = controller
+    rabbit_password = RABBIT_PASS
+
+Replace RABBIT_PASS with the password you chose for the guest account in RabbitMQ.
```
Comment (flopezag) [inline docker/muranoservice/README.md]: Take a look to the previous comment.

File: docker/muranoservice/README.md [hunk 12]
```diff
@@ -0,0 +1,237 @@
...
+
+In the [keystone_authtoken] and [ec2authtoken] sections, configure Identity service access:
```
Comment (flopezag) [inline docker/muranoservice/README.md]: In the [keystone_authtoken] and [ec2authtoken] sections, configure Identity service access replacing MURANO_PASS with the password you chose for the heat user in the Identity service:

File: docker/muranoservice/README.md [hunk 13]
```diff
@@ -0,0 +1,237 @@
...
+    auth_uri = http://controller:5000/v2.0
+    identity_uri = http://controller:35357
+    admin_tenant_name = service
+    admin_user = murano
+    admin_password = MURANO_PASS
+
+    [ec2authtoken]
+    ...
+    auth_uri = http://controller:5000/v2.0
+    Replace MURANO_PASS with the password you chose for the heat user in the Identity service.
```
Comment (flopezag) [inline docker/muranoservice/README.md]: Take a look to the previous comment

File: docker/muranoservice/README.md [hunk 14]
```diff
@@ -0,0 +1,237 @@
...
+
+In the murano section, it is required to specified the default region in a multiregion environment
+
+    [murano]
+
+    region_name_for_services = RegionOne
+
+In the rabbitmq section [rabbitmq], which represents the rabbit among murano-engine and virtual machines
```
Comment (flopezag) [inline docker/muranoservice/README.md]: among iff there are more than 2... :)

replace by between

add : at the end of the sentence.

File: docker/muranoservice/README.md [hunk 15]
```diff
@@ -0,0 +1,237 @@
...
+
+    [rabbitmq]
+    ...
+    host =130.206.84.6 (the rabbitmq host)
```
Comment (flopezag) [inline docker/muranoservice/README.md]: Just styling, add blank space after equal.

File: docker/muranoservice/README.md [hunk 16]
```diff
@@ -0,0 +1,237 @@
...
+    login = guest (The RabbitMQ login.)
+    password = guest (The RabbitMQ password)
+
+In the engine section, the folder where to configure classes required for [multiregion](http://docs.openstack.org/developer/murano/articles/multi_region.html).
```
Comment (flopezag) [inline docker/muranoservice/README.md]: confuse...
where you can find the required configured classes for...
or
where you can find the configured required classes for...

File: docker/muranoservice/README.md [hunk 17]
```diff
@@ -0,0 +1,237 @@
...
+
+    [engine]
+
+    class_configs = /etc/murano/class-configs
+
+In this folder, it is possible to specify a file for managing multi-region as io.murano.Environment.yaml, which the content
```
Comment (flopezag) [inline docker/muranoservice/README.md]: add :

File: docker/muranoservice/README.md [hunk 18]
```diff
@@ -0,0 +1,237 @@
...
+
+    regionConfigs:
+    Spain2:
+        agentRabbitMq:
+        host: RABBIT_HOST
```
Comment (flopezag) [inline docker/muranoservice/README.md]: Should be here an indent? I mean host, login and password should not be in the same column of agentRabbitMq

File: docker/muranoservice/README.md [hunk 19]
```diff
@@ -0,0 +1,237 @@
...
+    [murano]
+
+    region_name_for_services = RegionOne
+
+In the rabbitmq section [rabbitmq], which represents the rabbit among murano-engine and virtual machines
+
+    [rabbitmq]
+    ...
+    host =130.206.84.6 (the rabbitmq host)
+    login = guest (The RabbitMQ login.)
```
Comment (flopezag) [inline docker/muranoservice/README.md]: I think that you do not need the point at the end of it.

File: docker/muranoservice/README.md [hunk 20]
```diff
@@ -0,0 +1,237 @@
...
+    ...
+    external_network = public-ext-net-01 (the public network)
+    router_name = rt2-external (the router name to be created)
+
+
+### Populate the murano database
+
+Use the following command:
+
+    murano-db-manage upgrade
```
Comment (flopezag) [inline docker/muranoservice/README.md]: $ murano-db-manage upgrade

File: docker/muranoservice/README.md [hunk 21]
```diff
@@ -0,0 +1,237 @@
...
+
+### Import core library.
+It is required to obtain the core library which is in [murano repository](https://github.com/openstack/murano/tree/master/meta/io.murano)
+We need to execute the zip file for that folder and import in murano by:
+
+    murano package-import --exists-action u  --is-public io.murano.zip
```
Comment (flopezag) [inline docker/muranoservice/README.md]: $ murano package-import --exists-action u  --is-public io.murano.zip

File: docker/muranoservice/README.md [hunk 22]
```diff
@@ -0,0 +1,237 @@
...
+
+For obtaining the files for the murano core library, it is required to obtain the core library from murano Openstack repository plus the Fiware
```
Comment (flopezag) [inline docker/muranoservice/README.md]: OpenStack 👎 👎 👎 
FIWARE 🎱 

Come on...

File: docker/muranoservice/README.md [hunk 23]
```diff
@@ -0,0 +1,237 @@
...
+specific requirements. An example to do it can be:
+
+    mkdir /murano && cd /murano && git init && git remote add -f origin https://github.com/openstack/murano && git config core.sparseCheckout true && echo "meta/io.murano/" >> .git/info/sparse-checkout && git pull origin stable/newton
```
Comment (flopezag) [inline docker/muranoservice/README.md]: Add $ before each command

File: docker/muranoservice/README.md [hunk 24]
```diff
@@ -0,0 +1,237 @@
...
+
+    $ murano package-import /home/downloads/mysql.zip
+    Importing package com.example.databases.MySql
+    +---------------------------------+------+----------------------------+--------------+---------+
+    | ID                              | Name | FQN                        | Author       |Is Public|
+    +---------------------------------+------+----------------------------+--------------+---------+
+    | 83e4038885c248e3a758f8217ff8241f| MySQL| com.example.databases.MySql| Mirantis, Inc|         |
+    +---------------------------------+------+----------------------------+--------------+---------+
+
+Murano fiware packages are in the following [link](https://github.com/telefonicaid/fiware-enablers/tree/develop/murano-apps).
```
Comment (flopezag) [inline docker/muranoservice/README.md]: FIWARE

File: docker/muranoservice/README.md [hunk 25]
```diff
@@ -0,0 +1,237 @@
...
+
+## How to use muranoservice with Docker
+
+There are several options to use FIWARE_MURANO very easily using docker. These are (in order of complexity):
+
+- _"Have everything automatically done for me"_. See Section **1. The Fastest Way** (recommended).
```
Comment (flopezag) [inline docker/muranoservice/README.md]: Where is section 2?

And after it, where is the descripcion of section 2 and section 3?

File: docker/muranoservice/README.md [hunk 26]
```diff
@@ -0,0 +1,237 @@
...
+- _"Check the acceptance tests are running properly"_ or _"I want to check that my fiware-murano instance run properly"_ . See Section **3. Run Acceptance tests**.
+
+You do not need to do all of them, just use the first one if you want to have a fully operational Aiakos instance and maybe third one to check if your Aiakos instance run properly.
```
Comment (flopezag) [inline docker/muranoservice/README.md]: Copy/Paste ... Aiakos...

File: docker/muranoservice/README.md [hunk 27]
```diff
@@ -0,0 +1,237 @@
...
+    REPOSITORY          TAG                 IMAGE ID            CREATED              SIZE
+    muranoservice      latest              bd78d006c2ea        About a minute ago   480.8 MB
+    ...
+
+muranoservice image needs the PASSWORD variable to be defined, as well as RABBIT_HOST, RABBIT_LOGIN and RABBIT_PASSWORD for multi-region.
+In addition, it needs the docker mysql and rabbitmq alredy deployed.
+Thus, to deploy the contanair we need
+to execute the command:
+
+    docker run -p 8082:8082 -e PASSWORD=$PASSWORD -e RABBIT_HOST=RABBIT_HOST -e RABBIT_LOGIN=RABBIT_LOGIN -e RABBIT_PASSWORD=RABBIT_PASSWORD--link rabbit --link mysql -d fiware-murano
```
Comment (flopezag) [inline docker/muranoservice/README.md]: $ docker run -p 8082:8082 -e PASSWORD=$PASSWORD -e RABBIT_HOST=RABBIT_HOST -e RABBIT_LOGIN=RABBIT_LOGIN -e RABBIT_PASSWORD=RABBIT_PASSWORD--link rabbit --link mysql -d fiware-murano

File: docker/muranoservice/README.md [hunk 28]
```diff
@@ -0,0 +1,237 @@
...
+Take the Container ID and execute `docker stop b8e1de41deb5` or `docker kill b8e1de41deb5`. Note that you will lose any data that was being used in
+muranoservice using this method.
+
+However, there is a simpler way to deploy the container. That is docker-compose and it avoids to deploy containers previously and specifies the port for
+murano. It involves just exporting the following variables:
+
+    export PASSWORD=<OpenStack admin user password>
+    export RABBIT_HOST=<host where the rabbitmq is located>
+    export RABBIT_LOGIN=<login for the rabbitmq>
+    export RABBIT_PASSWORD=<password for the rabbitmq
```
Comment (flopezag) [inline docker/muranoservice/README.md]: do not forger '>'

File: docker/muranoservice/docker-compose.yml [hunk 29]
```diff
@@ -0,0 +1,32 @@
...
+     MYSQL_USER: mysql
+     MYSQL_PASSWORD: mysql
+     MYSQL_ROOT_PASSWORD: supersecret
+rabbit:
+   image: rabbitmq:3
+   container_name: rabbit
+   ports:
+   - 5672:5672
+murano:
+  image: murano-service
```
Comment (flopezag) [inline docker/muranoservice/docker-compose.yml]: I thought that the name of the image was muranoservice, at least it is written in the documentation in this way, isn't?

File: docker/muranoservice/Dockerfile (hunk 1 above)
Comment (hmunfru) [reply docker/muranoservice/Dockerfile:8]: It is in the README file

File: docker/muranoservice/README.md (hunk 2 above)
Comment (hmunfru) [reply docker/muranoservice/README.md]: apt-get

File: docker/muranoservice/README.md (hunk 7 above)
Comment (hmunfru) [reply docker/muranoservice/README.md]: ok

File: docker/muranoservice/Dockerfile (hunk 1 above)
Comment (flopezag) [reply docker/muranoservice/Dockerfile:8]: I mean, I want that the Dockerfile starts with two lines in which you write how to create the image and how to launch the container. Just the docker commands.

**Comment by flopezag** [comment]
LGTM

**Pull request merged**